anime_downloader/commands/test.py

```diff
@@ -3,66 +3,178 @@
-        else:
-            p, f = '✅ ', '❌ '
-        thread.join(timeout=10)
-        if not thread.is_alive():
-            if not thread.exception:
-                # echo(click.style('Works ', fg='green') + site)
-                echo(click.style(p, fg='green') + thread.site)
+    for i, thread in enumerate(threads):
+        try:
+            click.echo(f"[{i+1} of {len(threads)}] Searching ", nl=False)
+            click.secho(f"{thread.provider}", nl=False, fg="cyan")
+            click.echo(f"... (CTRL-C to stop) : ", nl=False)
+            thread.join(timeout=timeout)
+            if not thread.is_alive():
+                if not thread.exception:
+                    if thread.search_result:
+                        if not no_fuzzy:
+                            ratio = fuzz.token_set_ratio(anime.lower(), thread.search_result.lower())
+                        else:
+                            ratio = 100
+                        if ratio > 50:
+                            matches.append([thread.provider, thread.search_result, ratio])
+                            click.secho(p + "Works, anime found.", fg="green")
+                            if not full_search:
+                                click.echo(f"\n- - -{thread.provider}- - -\n\n{thread.search_result}")
+                                confirm = click.confirm(f"Found anime in {thread.provider}. Keep seaching? (use -f / --full-search to disable this prompt)", default=True)
+                                if not confirm:
+                                    break
+                        else:
+                            click.secho(p + "Works, anime not found.", fg="yellow")
+                    else:
+                        click.secho(p + "Works, anime not found.", fg="yellow")
+                else:
+                    logging.debug('Error occurred during testing.')
+                    logging.debug(thread.exception)
+                    if thread.search_result:
+                        click.secho(f + "Not working: anime found, extraction failed.", fg="red")
+                    else:
+                        click.secho(f + "Not working.", fg="red")
             else:
-                logging.debug('Error occurred during testing')
-                logging.debug(thread.exception)
-                echo(click.style(f, fg='red') + thread.site)
-        else:
-            logging.debug('timeout during testing')
-            echo(click.style(f, fg='red') + thread.site)
+                logging.debug('Timeout during testing.')
+                click.secho(f + "Not working: Timeout. Use -t to specify longer waiting period.", fg="red")
+
+        except KeyboardInterrupt:
+            skip = click.confirm(f"\nSkip {thread.provider} and continue searching? (Press enter for Yes)", default=True)
+            if not skip:
+                break
+
+    if not no_results:
+        click.echo("\n" + util.format_matches(matches))
```

Comment:
Maybe instead of trying to format the matches, just print `Anime Test Finished` because it will overtake the screen with a broken table without it. change it like this:
```click.echo("\n" + "Anime test finished")```
or if you still want it, make it a switch to enable it. Keep in mind the formatting of the table would have to be fixed.